implementation/leprechaun/src/shared/api/generic-paginated.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class PaginatedDto<TDto> {
+  readonly data: TDto[];
+
+  @ApiProperty({ description: 'Total number of items', example: 420, minimum: 0 })
+  readonly total: number;
+  @ApiProperty({ description: 'Number of items per page', example: 30, minimum: 1, maximum: 100, default: 10 })
+  readonly size: number;
+  @ApiProperty({ description: 'Current page number', example: 12, minimum: 0, default: 0 })
+  readonly page: number;
+
+  constructor(total, size, page, data) {
+    this.total = total;
+    this.size = size;
+    this.page = page;
+    this.data = data;
+  }
+
+  static fromEntities<TDto>(total: number, size: number, page: number, entities: TDto[]): PaginatedDto<TDto> {
+    return new PaginatedDto<TDto>(total, size, page, entities);
+  }
+
+  static sanitizePagination(size: number, page: number): { size: number; page: number } {
+    return {
+      page: Math.max(0, page),
+      size: Math.min(100, Math.max(1, size)),
+    };
+  }
```

Comment:
It is, but I keep it here because whenever I use `PaginatedDto`, I have to sanitize parameters, so it is a little bit cleaner than creating new file with 5 lines. I maybe will in the future make a `Pipe` for these.